File: src/modules/company/services/company_find_one_service.py
from src.modules.company.models import Company
from src.shared.types.user_request import UserRequest
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.validators.user_company_validator import (
    user_company_validator,
)


class CompanyFindOneService:
    @classmethod
    def execute(self, id: int, user_request: UserRequest):

        company = Company.objects.filter(id=id, deleted_at=None).first()

        if company is None:
            raise NotFoundException("Empresa não encontrada!")

        user_company_validator(id, user_request)

        return company

File: src/modules/company/services/company-find-one.service.ts
import { prisma } from '../../../infra/database/prisma.service';
import { NotFoundException } from '../../../shared/exceptions';
import { UserRequest } from '../../../shared/types/user-request';
import { userCompanyValidator } from '../../../shared/validators/user-company.validator';
import { CompanyEntity } from '../domain/entities/company.entity';

class CompanyFindOneService {
  async execute(id: number, userRequest: UserRequest): Promise<CompanyEntity> {
    const company = await prisma.company.findFirst({
      where: {
        id: id,
        deletedAt: null,
      },
      include: {
        address: true,
        avatar: true,
      },
    });

    if (!company) throw new NotFoundException('Empresa não encontrada!');

    await userCompanyValidator(id, userRequest);

    return new CompanyEntity(company);
  }
}

export const companyFindOneService = new CompanyFindOneService();
